pkg/object/httpserver/mux.go

```diff
@@ -569,25 +569,16 @@ func (mi *muxInstance) serveHTTP(stdw http.ResponseWriter, stdr *http.Request) {
 }
 
 func (mi *muxInstance) search(req *httpprot.Request) *route {
-	headerMismatch, methodMismatch, queryMismatch := false, false, false
+	headerMismatch, methodMismatch, queryMismatch, clientIPMismatch := false, false, false, false
 
 	ip := req.RealIP()
 
 	// The key of the cache is req.Host + req.Method + req.URL.Path,
 	// and if a path is cached, we are sure it does not contain any
-	// headers or any queries.
+	// headers,any queries and any ipFilters.
 	r := mi.getRouteFromCache(req)
 	if r != nil {
-		if r.code != 0 {
-			return r
-		}
-		if r.path.ipFilterChain == nil {
-			return r
-		}
-		if r.path.ipFilterChain.Allow(ip) {
-			return r
-		}
-		return forbidden
+		return r
```

Comment:
this change hasn't covered the case at L585, which result in a blocked IP being allowed.

Reply:
Actually, if a route can be put into cache, there are no ipFilterChain at all, so this case won't happen.

So it is a problem here, if there is a ipFilter at muxInstance, there will be no cache.  :(

I'm trying to get a better way to do that. WIP.